File: sketches/sketches.model/src/main/java/edu/kit/kastel/lissa/sketches/model/impl/AbstractElement.java
package edu.kit.kastel.lissa.sketches.model.impl;

import java.io.Serializable;
import java.util.HashMap;
import java.util.Map;

import edu.kit.kastel.lissa.sketches.model.elements.ISketchElement;

public abstract class AbstractElement implements ISketchElement {
	private static final long serialVersionUID = -7524099797275769182L;

	@SuppressWarnings("unused")
	private transient Map<String, Serializable> data = new HashMap<>();

	private String name;
	private double confidence;

	protected AbstractElement() {
		// NOP
	}

	public void setName(String name) {
		this.name = name;
	}

	public void setConfidence(double confidence) {
		this.confidence = confidence;
	}

	@Override
	public double getCurrentConfidence() {
		return this.confidence;
	}

	@Override
	public String getName() {
		return this.name;
	}

	public Map<String, Serializable> getRawData() {
		return this.data;
	}

	@Override
	public String toString() {
		return String.format("%s [name=%s, confidence=%s]", //
				this.getClass().getSimpleName(), this.getName(), this.getCurrentConfidence());
	}

}

File: sketches/model/src/main/kotlin/edu/kit/kastel/lissa/sketches/model/impl/AbstractElement.kt
package edu.kit.kastel.lissa.sketches.model.impl

import edu.kit.kastel.lissa.sketches.model.elements.ISketchElement
import java.util.*

abstract class AbstractElement protected constructor() : ISketchElement {
    @Transient
    val rawData: MutableMap<String, java.io.Serializable> = mutableMapOf()

    private var nameValue: String = ""
    private var currentConfidenceValue = 0.0
    private var elementId = UUID.randomUUID().toString()

    override fun name(): String = nameValue
    override fun setName(name: String) {
        nameValue = name
    }

    override fun currentConfidence() = currentConfidenceValue
    override fun setCurrentConfidence(confidence: Double) {
        currentConfidenceValue = confidence
    }

    override fun id(): String = this.elementId
    override fun setId(id: String) {
        this.elementId = id
    }

    override fun toString(): String {
        return String.format(
            "%s [name=%s, confidence=%s]", //
            this.javaClass.simpleName,
            nameValue,
            currentConfidenceValue
        )
    }
}
